samples/sample_decode/src/sample_decode.cpp

```diff
@@ -749,6 +749,14 @@ mfxStatus ParseInputString(msdk_char* strInput[], mfxU8 nArgNum, sInputParams* p
     }
 #endif
 
+#if defined(_WIN32) || defined(_WIN64)
+    if (pParams->videoType == MFX_CODEC_AV1 && pParams->memType == D3D9_MEMORY)
```

Comment:
It will fail even with system memory but if MSDK session was initialized dx9 path. Can you please move this check to the place right after `mfxInit(); mfxQueryIMPL(&impl_type);`  :
e.g.
```
mfxInit()
mfxQueryIMPL(&impl_type);
if (AV1 && (impl_type | DX9)) {
  printf("blah blah blah\n");
  return error;
}
```